lib/GILGen/TypeTranslator.cpp

```diff
@@ -0,0 +1,118 @@
+#include "TypeTranslator.hpp"
+#include <llvm/Support/Casting.h>
+
+namespace glu::gilgen {
+
+gil::Type TypeTranslator::visitIntTy(types::IntTy *type)
+{
+    // Calculate size in bytes from bit width
+    unsigned size = type->getBitWidth() / 8;
```

Comment:
Well a 15 bit int should be size 2
```suggestion
    unsigned size = (type->getBitWidth() + 7) / 8;
```